cmd/serverless/main.go

```diff
@@ -132,13 +109,12 @@ func runAgent(stopCh chan struct{}) (serverlessDaemon *daemon.Daemon, err error)
 		if processError != nil {
 			log.Errorf("Can't process events: %s", processError)
 		}
-		return nil, nil
+		return nil
 	}
 
 	// immediately starts the communication server
-	serverlessDaemon = daemon.StartDaemon(httpServerAddr)
-	serverlessDaemon.ExecutionContext.SetInitializationTime(startTime)
-	err = serverlessDaemon.ExecutionContext.RestoreCurrentStateFromFile()
+	serverlessDaemon.Start(startTime)
+	err = serverlessDaemon.GetExecutionContext().RestoreCurrentStateFromFile()
```

Comment:
Would it make sense to move  line 117-122 to `Start` method?

Reply:
Agreed! I moved this block of code to inside `Start`